### ISSUE TITLE
Twiml Apps

### ISSUE THREAD

**Issue by fawcilize** [RJPearson94/terraform-provider-twilio#33]
## Is your feature request related to a problem

It's not clear whether it's possible to create a Twiml App

## Describe the solution you'd like
A way to make a Twiml App

```hcl
resource "twilio_twiml_app" "resource_name" {
  friendly_name = ""
  voice {
    request_url = ""
    fallback_url = ""
    status_callback_url = ""
    caller_name_lookup = "disabled|enabled"
  }

  messaging {
    request_url = ""
    fallback_url = ""
    status_callback_url = ""
  }
}
```

## Additional context

The Twiml Apps appear under the Phone Number and Voice sections. It's unclear where the ownership lies.

**Comment by RJPearson94** [comment]
Hi @fawcilize,

Thanks for checking out the provider and for the feature request. 

At the moment the provider doesn't support Applications (which is what TwiML apps are) but I can look at adding support. I will add it to the roadmap for the next release (v0.12.0).

As TwiML apps are Applications under the hood, I'm leaning towards using a more generic resource name instead of a TwiML specific name. Any feedback is welcome on this though

I take a look at adding support

**Comment by RJPearson94** [comment]
Hi @fawcilize 

Apologies for the delay in delivering this feature request and thank you for your patience.

The code is currently in the main branch and is on track to be released in `v0.12.0` (hopefully should be released on the 13th June 2021). I will update this issue.

I have gone with your original name suggestion and I have added it under the twiml subcategory, as I am considering adding the ability to generate TwiML voice and message responses in future releases. The docs for the resource can be seen [here](https://github.com/RJPearson94/terraform-provider-twilio/blob/main/docs/resources/twiml_app.md)

I will update this issue when the change is released.

Thanks again for trying out the provider and for the feature request and any feedback is greatly appreciated.

**Comment by RJPearson94** [comment]
This has now been released as part of v0.12.0. This release is now available in the Terraform Registry.

Please update your Terraform configuration to include the following

```
terraform {
  required_providers {
    twilio = {
      source = "RJPearson94/twilio"
      version = "0.12.0"
    }
  }
}
```
